public/angular/js/controllers/fetching.js

```diff
@@ -3,32 +3,25 @@ angular.module('Aggie')
 .controller('FetchingController', [
   '$scope',
   'Fetching',
-  '$timeout',
-  function($scope, Fetching, $timeout) {
+  'FlashService',
+  function ($scope, Fetching, flash) {
 
     Fetching.get(function success(data) {
       $scope.on = data;
     }, failure);
 
-    $scope.toggle = function (x) {
-      $scope.on = x;
+    $scope.toggle = function(x) {
+      Fetching.set(x, success(x), failure);
     };
 
-    $scope.save = function () {
-      $scope.loading = true;
-      Fetching.set($scope.on, success, failure);
+    function success(data) {
+      $scope.on = data;
+      flash.setNoticeNow('Fetching has been successfully switched.');
```

Comment:
I don't think you need to say this explicitly just for flipping a switch. The assumption will be that it worked. Just showing an error in case of failure is fine.